### PR TITLE
CDAP-15109 Use all active run states to determine concurrency constraint for schedules

### PR DESCRIPTION
**Pull request by poornachandra**
JIRA - https://issues.cask.co/browse/CDAP-15109
Build - https://builds.cask.co/browse/CDAP-DUT6937

### CODE REVIEW

**Comment by nitinmotgi** [comment]
Small comment

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/constraint/ConcurrencyConstraintTest.java [hunk 1]
```diff
@@ -129,6 +135,16 @@ public void testMaxConcurrentRuns() {
                     AppFabricTestHelper.createSourceId(++sourceId));
       assertSatisfied(true, concurrencyConstraint.check(schedule, constraintContext));
 
+      // add a run in provisioning state, constraint will not be satisfied since active runs increased to 2
+      ProgramRunId pid4 = WORKFLOW_ID.run(RunIds.generate().getId());
+      setStartAndRunning(store, pid4);
```
Comment (anew) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/constraint/ConcurrencyConstraintTest.java]: setProvisioning?

Comment (poornachandra) [reply cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/constraint/ConcurrencyConstraintTest.java]: Fixed

**Review by anew — APPROVED** [review]

**Comment by poornachandra** [comment]
Build is green, merging...

**Pull request merged**